TFUtil.py

```diff
@@ -393,19 +393,37 @@ def copy_with_feature_dim_axis(self, feature_dim_axis):
     other_special_axes = data.get_special_axes_dict(
       counted_with_batch_dim=True, only_available=True, include_batch_dim_axis=True)
     other_special_axes.pop("feature_dim_axis", None)
+    old_feature_axis = data.feature_dim_axis
     data.feature_dim_axis = feature_dim_axis
     for k, a in other_special_axes.items():
-      setattr(data, k, a if (a < feature_dim_axis) else (a + 1))
+      if a == feature_dim_axis:
+        if a < old_feature_axis:
+          a += 1
+        else:
+          a -= 1
+      elif a > feature_dim_axis:
+        if a < old_feature_axis:
+          a += 1
+      else:
+        if a > old_feature_axis:
+if old_feature_axis <= a < feature_dim_axis:
+  a -= 1
+elif feature_dim_axis <= a < old_feature_axis:
+  a += 1
+      setattr(data, k, a)
```

Comment:
I'm not sure what happened here. There is still your old code, but then also my suggested code, but with a wrong indentation. Please fix that.